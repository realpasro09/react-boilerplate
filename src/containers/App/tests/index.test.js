import React from 'react';
import {App, mapStateToProps} from '../index';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router';

describe('<App />', () => {
  it('Expect to be render correctly', () => {
    const props = {
      buildInformation: {
        deploymentEnvironment: 'test',
        frontEndBuildNumber: 123,
      },
      sampleData: [{id: 1, name: 'test'}],
      actions: {
        getSampleData: jest.fn(),
      },
    };
    const wrapper = renderer.create(
      <MemoryRouter>
        <App {...props} />
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });
  describe('mapStateToProps', () => {
    it('should return app buildinformation object', () => {
      expect(
        mapStateToProps({
          app: {
            buildInformation: {
              deploymentEnvironment: 'test',
              frontEndBuildNumber: 123,
            },
          },
        }).buildInformation
      ).toEqual({
        deploymentEnvironment: 'test',
        frontEndBuildNumber: 123,
      });
    });
  });
});
