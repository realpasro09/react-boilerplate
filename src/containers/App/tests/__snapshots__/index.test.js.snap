// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Expect to be render correctly 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <h1>
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
      Welcome to React
    </h1>
  </div>
  <div
    className="jumbotron"
  >
    <p
      className="App-intro"
    >
      To get started, edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
  </div>
  <div>
    <h1>
      SAGA SAMPLE
    </h1>
    <div>
      test
    </div>
  </div>
  <div
    className="well"
  >
    <p>
      Front-End Version 
      test
       
      123
    </p>
    <p>
      Back-End Version (get from the api) 
    </p>
  </div>
</div>
`;
